subsys/net/lib/nrf_cloud/src/nrf_cloud_cell_pos.c

```diff
@@ -0,0 +1,312 @@
+	    !cJSON_AddStringToObject(resp_obj, CELL_POS_JSON_MSG_TYPE_KEY, msg_type)) {
+		cJSON_Delete(resp_obj);
+		resp_obj = NULL;
+	}
+
+	return resp_obj;
+}
+
+static int json_format_data_obj(cJSON *const data_obj,
+				const struct modem_param_info *const modem_info)
+{
+	__ASSERT_NO_MSG(data_obj != NULL);
+	__ASSERT_NO_MSG(modem_info != NULL);
+
+	if (!cJSON_AddNumberToObject(data_obj, CELL_POS_JSON_MCC_KEY,
+		modem_info->network.mcc.value) ||
+	    !cJSON_AddNumberToObject(data_obj, CELL_POS_JSON_MNC_KEY,
+		modem_info->network.mnc.value) ||
+	    !cJSON_AddNumberToObject(data_obj, CELL_POS_JSON_AREA_CODE_KEY,
+		modem_info->network.area_code.value) ||
+	    !cJSON_AddNumberToObject(data_obj, CELL_POS_JSON_CELL_ID_KEY,
+		(uint32_t)modem_info->network.cellid_dec) ||
+	    !cJSON_AddNumberToObject(data_obj, CELL_POS_JSON_PHYCID_KEY, 0)) {
+		return -ENOMEM;
+	}
+
+	return 0;
+}
+
+static int json_add_modem_info(cJSON *const data_obj)
+{
+	__ASSERT_NO_MSG(data_obj != NULL);
+
+	struct modem_param_info modem_info = {0};
+	int err;
+
+	err = get_modem_info(&modem_info);
+	if (err) {
+		return err;
+	}
+
+	return json_format_data_obj(data_obj, &modem_info);
+}
+
+static int json_send_to_cloud(cJSON *const cell_pos_request)
+{
+	__ASSERT_NO_MSG(cell_pos_request != NULL);
+
+	char *msg_string;
+	int err;
+
+	msg_string = cJSON_PrintUnformatted(cell_pos_request);
+	if (!msg_string) {
+		LOG_ERR("Could not allocate memory for Cell Pos request message");
+		return -ENOMEM;
+	}
+
+	LOG_DBG("Created Cell Pos request: %s", log_strdup(msg_string));
+
+	struct nct_dc_data msg = {
+		.data.ptr = msg_string,
+		.data.len = strlen(msg_string)
+	};
+
+	err = nct_dc_send(&msg);
+	if (err) {
+		LOG_ERR("Failed to send Cell Pos request, error: %d", err);
+	} else {
+		LOG_DBG("Cell Pos request sent");
+	}
+
+	k_free(msg_string);
+
+	return err;
+}
+
+bool json_item_string_exists(const cJSON *const obj, const char *const key, const char *const val)
+{
+	__ASSERT_NO_MSG(obj != NULL);
+	__ASSERT_NO_MSG(key != NULL);
+
+	char *str_val;
+	cJSON *item = cJSON_GetObjectItem(obj, key);
+
+	if (!item) {
+		return false;
+	}
+
+	if (!val) {
+		return cJSON_IsNull(item);
+	}
+
+	str_val = cJSON_GetStringValue(item);
+	if (!str_val) {
+		return false;
+	}
+
+	return (strcmp(str_val, val) == 0);
+}
+
+static int json_parse_cell_location(const cJSON *const cell_pos_obj,
+				    struct nrf_cloud_cell_pos_result *result)
+{
+	__ASSERT_NO_MSG(cell_pos_obj);
+
+	cJSON *lat, *lon, *unc;
+
+	lat = cJSON_GetObjectItem(cell_pos_obj, CELL_POS_JSON_CELL_LOC_KEY_LAT);
+	lon = cJSON_GetObjectItem(cell_pos_obj, CELL_POS_JSON_CELL_LOC_KEY_LON);
+	unc = cJSON_GetObjectItem(cell_pos_obj, CELL_POS_JSON_CELL_LOC_KEY_UNCERT);
+
+	if (!cJSON_IsNumber(lat) || !cJSON_IsNumber(lon) || !cJSON_IsNumber(unc)) {
+		LOG_DBG("Expected items not found in cell-based location msg");
+		return 1; /* 1: cell-based location not found */
```

Comment:
this should be changed back to the original code:
`return -EBADMSG;`

if this function has been called, the data *should* contain cell pos data.

Reply:
Restored